### PR TITLE
[9/20] MWPW-136568: Table - Mobile issue with new milo events

### PR DESCRIPTION
**Pull request by seanchoi-dev**
Fixing bug for table mobile logic doesn't work at all with new milo eventListener.
This issue is easily happening specially when there are more than one table in a page.

Resolves: [MWPW-136568](https://jira.corp.adobe.com/browse/MWPW-136568)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/docs/library/blocks/table?martech=off
- After: https://mwpw-136568--milo--adobecom.hlx.page/docs/library/blocks/table?martech=off

**Issue page on production**
- Before: https://main--bacom--adobecom.hlx.page/products/marketo/pricing
- After: https://main--bacom--adobecom.hlx.page/products/marketo/pricing?milolibs=mwpw-136568

*note: this fix is needed for CC release on 9/20 according to Ruchika Sinha.

### CODE REVIEW

**Comment by Ruchika4** [comment]
Thanks for the fix @seanchoi-dev . However it removes all the checkmarks from the table and shows and empty table. Could you please check. Thanks.
Before: https://main--cc--adobecom.hlx.page/drafts/ruchika/3d-animation-software
Screen Shot 2023-09-18 at 11 07 29 AM

After: https://main--cc--adobecom.hlx.page/drafts/ruchika/3d-animation-software?milolibs=MWPW-136568
Screen Shot 2023-09-18 at 11 09 08 AM

File: libs/blocks/table/table.js [hunk 1]
```diff
@@ -414,6 +414,7 @@ function applyStylesBasedOnScreenSize(table, originTable) {
 
 export default function init(el) {
   el.setAttribute('role', 'table');
+  if (el.parentElement.classList.contains('section')) el.parentElement.classList.add(`table-${el.classList.contains('merch') ? 'merch-' : ''}section`);
```
Comment (chrischrischris) [inline libs/blocks/table/table.js:417]: line > 100

**Review by chrischrischris — APPROVED** [review]

**Comment by Ruchika4** [comment]
> Thanks for the fix @seanchoi-dev . However it removes all the checkmarks (we are using checkmark icons from library) from the table and shows and empty table. Could you please check. Thanks. Before: https://main--cc--adobecom.hlx.page/drafts/ruchika/3d-animation-software Screen Shot 2023-09-18 at 11 07 29 AM
> 
> After: https://main--cc--adobecom.hlx.page/drafts/ruchika/3d-animation-software?milolibs=MWPW-136568 Screen Shot 2023-09-18 at 11 09 08 AM

This is fixed with latest commit. Thanks Sean.

**Comment by Ruchika4** [comment]
@seanchoi-dev could you please look at the unit test case failures.

File: libs/blocks/table/table.css [hunk 2]
```diff
@@ -454,6 +397,78 @@ header.global-navigation {
   margin-right: 16px;
 }
 
+/* Start tablet styles */
+@media (max-width: 899px) {
```
Comment (overmyheadandbody) [inline libs/blocks/table/table.css:401]: Is there a specific reason for going desktop-first and not mobile-first? (I get that these were just copied over for this PR, but still wondering about the decision). I also see on L324 that `min-width` is used, I think mixing the two will lead to confusion.

File: libs/blocks/table/table.css [hunk 3]
```diff
@@ -454,6 +397,78 @@ header.global-navigation {
...
+    background-color: var(--color-gray-100);
+  }
+
+  .table .row-heading .col:nth-child(n+1) {
+    padding: 20px;
+  }
+
+  .table:not(.highlight):not(.merch) .col-2.col-heading {
+    border-top-left-radius: 16px;
+  };
```
Comment (overmyheadandbody) [inline libs/blocks/table/table.css:434]: Semicolon should be removed

**Review by overmyheadandbody — COMMENTED** [review]
I see UT are failing, that should be addressed

**Review by Ruchika4 — APPROVED** [review]
Thanks for the fix @seanchoi-dev

**Pull request merged**